### PR TITLE
Addresses issue #4362

### LINKED ISSUE #4362: GitHub tags are only read up to the first `-` character, breaking tags such as `1.0.0-rc1`
### The problem

The software update plugin does a load of parsing to compare the GitHub tag version to the one from the installed Python packages. I unwrapped some of it to figure out what was going on here.

We have a tag on GitHub with the name `1.0.0-rc1`, which according to [PEP440](https://www.python.org/dev/peps/pep-0440/#pre-release-separators) is allowed but would be normalized to `1.0.0rc1`, which is how it comes through if this tag is set in setup.py.

The plugin runs it through (as far as I could trace) `octoprint.plugins.softwareupdate.version_checks.github_release.get_comparable_version`. This calls some functions in `octoprint.util.version` including `normalize_version`, which only reads the tag up to and including the `-`, stripping it down to just `1.0.0` in this case, which would be seen as 'ahead' of `1.0.0rc1` of course.

Some testing to demonstrate, the first line is the problem one:
```python
>>> get_comparable_version("1.1.1-rc1")
<Version('1.1.1')>
>>> get_comparable_version("1.1.1rc1")
<Version('1.1.1rc1')>
>>> get_comparable_version("1.1.1.rc1")
<Version('1.1.1rc1')>
>>> get_comparable_version("1.1.1_rc1")
<Version('1.1.1rc1')>
```

They should all be parsed equally according to PEP440.

This resulted in an update loop where `1.0.0-rc1` was set the same in both `setup.py` and the GitHub tag, but it became GitHub tag = 1.0.0 and setup.py = 1.0.0rc1 when parsed and the comparison would always show an update available.

**TLDR**; everything after the `-` in a GitHub tag is ignored, when it should be included.

Discussion on discord: https://discord.com/channels/704958479194128507/708230829050036236/928402397435420775

### Did the issue persist even in safe mode?

Yes, it did persist

### If you could not test in safe mode, please state why

NA

### Version of OctoPrint

1.7.2

### Operating system running OctoPrint

All

### Printer model & used firmware incl. version

All

### Browser and version of browser, operating system running browser

All

### Checklist of files to include below

- [X] Systeminfo Bundle (always include!)
- [ ] Contents of the JavaScript browser console (always include in cases of issues with the user interface)
- [ ] Screenshots and/or videos showing the problem (always include in case of issues with the user interface)
- [ ] GCODE file with which to reproduce (always include in case of issues with GCODE analysis or printing behaviour)

### Additional information & file uploads

_No response_

### PR DESCRIPTION
**Pull request by GonzoDMX**
#### What does this PR do and why is it necessary?
Fixes the normalize_version function in src/util/version.py so that it returns the correct version when the input version string is a release candidate containing the characters "-rc".

#### How was it tested? How can it be tested by the reviewer?
Tested in OctoPrint dev environment.

#### What are the relevant tickets if any?
Fixes #4362

### CODE REVIEW

File: src/octoprint/util/version.py [hunk 1]
```diff
@@ -208,6 +208,9 @@ def is_prerelease(version_string):
 
 
 def normalize_version(version):
+    if "-rc" in version:
+        version = version.replace("-rc", "rc")
+
```
Comment (cp2004) [inline src/octoprint/util/version.py:213]: This only covers the very specific case of having `-rc`.

By PEP440 version spec, https://www.python.org/dev/peps/pep-0440/#pre-release-separators, there could be `-a`, `-b`, `-c`, `-alpha`, `-beta`, `-pre`, `-preview`, `-rc` and possibly more I've missed off.

Probably better to either leave the `-` in (it should be parsed properly by `pkg_resources` or whatever's doing it) or always remove it.

**Review by cp2004 — CHANGES_REQUESTED** [review]

File: src/octoprint/util/version.py (hunk 1 above)
Comment (GonzoDMX) [reply src/octoprint/util/version.py:213]: I agree this is very naive attempt at fixing the issue, initially I had proposed a more general solution along those lines however, leaving the `-` will fail the unit test and simply removing the `-` for all cases will fail the unit test. I changed to commit to see if this would pass.
I believe leaving it in or simply removing it for all cases should work fine but it will fail a unit test... so maybe that's what needs to change?

Comment (GonzoDMX) [reply src/octoprint/util/version.py:213]: So more details on the unitest, it tests the string "1.6.0-dev123" and expects "1.6.0". Problem is I believe `-dev` is valid in PEP440.
So the best solution is probably to correct the test and leave in the `-`

**Comment by GonzoDMX** [comment]
I've updated the PR, upon further assessment of the problem I believe removing the `-` char is redundant. The version is passed immediately after to pkg_resources.parse_version() which does this for us already.
I've updated the unit test for the function normalize_versions() to remove the case testing the `-` character.

**Review by cp2004 — APPROVED** [review]
I think that's the right way to go 👍
